### ISSUE TITLE
Screen reader improvements - Assets and Categories sections

### ISSUE THREAD

**Issue by ajanec01** [craftcms/cms#7417]
###  Issues found in the "tree view navigation" and filters section

- ~~Additional description in the new heading section and in the asset/ categories fields would benefit from `aria-describedby`~~ - fixed
- Additional description in the new heading dialog would benefit from `aria-describedby`
![additional-description](https://user-images.githubusercontent.com/47136246/104253527-96e93880-546c-11eb-8b50-4caba63c282e.png)
- Not sure if  the folder view should be a navigation panel- tree view seems more appropriate at least for the Assets section where there is a clear indication of sub sectioning; worth having a look at aria [tree view example 1 ](https://www.w3.org/TR/wai-aria-practices-1.2/examples/treeview/treeview-1/treeview-1a.html )and [example 2-](https://www.w3.org/TR/wai-aria-practices-1.2/examples/treeview/treeview-1/treeview-1b.html)
- The all/trash and sort by buttons could provide a bit more context to screen reader users; consider adding a visually hidden text
![two-listboxes-all](https://user-images.githubusercontent.com/47136246/104253781-3f979800-546d-11eb-8746-da2c575eec4f.png)
- All/trashed filter should have only one listbox
- Use input type search for the searchbox
- Custom fields for assets are announced only once because they have the same `id` as the value of `aria-controls` (menu.Garnish763093901-option-0)
![aria-controls](https://user-images.githubusercontent.com/47136246/104253825-5938df80-546d-11eb-9c04-600cdd869673.png)
- There are two listboxes inside the sort by filter, for values and for ascending/descending- using one listbox and grouping content would make it more understandable- see [aria listbox example ](https://www.w3.org/TR/wai-aria-practices-1.2/examples/listbox/listbox-grouped.html)
- Action button that is displayed after a checkbox has been selected indicates wrong `aria-exapnded`  in response to the Esc key
![action-button](https://user-images.githubusercontent.com/47136246/104254088-f8f66d80-546d-11eb-8350-ff3bb7276ed1.png)
- The edit pop-up would benefit from `role=”dialog”` and an accessible name
![Edit-popup](https://user-images.githubusercontent.com/47136246/104254149-175c6900-546e-11eb-8759-44793aa988c2.png)
- Checkboxes in customize dialog should be grouped and should have the legend associated
![checkbox-group](https://user-images.githubusercontent.com/47136246/104254397-9c478280-546e-11eb-95ef-e5bb78670159.png)
- Elements that reorder checkboxes and folders are failing name, role, value (customise dialog) 
- Folders in the customise dialog are failing the name, role, value criterion
![reorder-folders](https://user-images.githubusercontent.com/47136246/104254373-8fc32a00-546e-11eb-8e71-dbfcdb5878b8.png)
- The expandable button next to save and new category buttons does not have a label; it also doesn’t convey the appropriate state when dismissed using the Esc key
- Two listboxes for the expandable button next to save

### Additional info

- Craft version: 3.5.18

**Comment by ajanec01** [comment]
With the exception of the last two points and the edit pop-up, all other issues apply to the Users page.

**Comment by gcamacho079** [comment]
I’m closing this issue as many issues have been addressed as of 5.2.8, and some are stale due to UI changes.

## Resolved:
- Instructions inside the dialog are referenced via `aria-describedby` (Resolved in #15395)
- The Status filter (i.e., All, Trashed)All/trashed filter uses only one listbox
- The edit pop-up (which is now a slideout) has a `role="dialog"`
- The sort and filter UI has been updated to use a disclosure menu with separate inputs for sort attribute and direction, so the two issues of (a) duplicate IDs and (b) listbox grouping have been eliminated.
- The Action button `aria-expanded` values have been fixed to reflect the open/closed state
- Split buttons throughout the control panel have been updated to have the accessible name of “More actions”; they follow the APG guidelines for keyboard interaction and ARIA states.
- Checkboxes in the Customize Sources dialog are grouped with associated `fieldset` and `legend` elements. The instructions are also referenced from the `fieldset` via `aria-describedby`
- Source buttons in the navigation now have appropriate NRV

## Outstanding
The primary outstanding issue from this ticket — reorder buttons not having appropriate NRV — has been moved into our issue tracking.
